beam/core/src/test/java/cz/o2/proxima/beam/direct/io/CommitLogReadTest.java

```diff
@@ -54,26 +54,36 @@
 
   @Test(timeout = 30000)
   public void testReadingFromCommitLog() {
```

Comment:
how is the new functionality (confirms) stress by these tests? I probably I'm missing something obvious here

Reply:
here:
```java
ListCommitLog commitLog = ListCommitLog.ofNonExternalizable(data, direct.getContext());
```
That creates the new `ListCommitLog`, which mimics non-externalizable offsets (that are offsets that cannot be persisted in external system - checkpoint - and be sure they represent the same element even upon recovery). Commit-logs with "non-externalizable offsets" (e.g. Google PubSub) rely heavily on ack() and nack() of elements to ensure at-least-once semantics (typically not exactly once-semantics, because when consumer consumes element and does neither ack() nor nack() it until timeout, the element is resend to another (or the same) consumer).

Reply:
thanks for the explanation ;) This sheds some light to ListCommitLog internals 👍

Reply:
Can you please add this explation to `cz.o2.proxima.direct.storage.ListCommitLog#ofNonExternalizable` javadoc?

Reply:
ok